docs/challenges/language-support/cpp.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: "C++"
+description: "Learn how to solve CodeCrafters challenges in C++"
+---
+
+# Local development
+
+To run the code for your challenge locally, you'll need `cmake` & `make` installed.
+
+The script for your challenge (like `./your_sqlite3.sh`) will automatically compile your code using the rules defined in `CMakeLists.txt` before executing it. It'll
+look something like this:
+
+```sh
+#!/bin/sh
+
+set -e
+cmake $(dirname $0) > /dev/null
+make > /dev/null
+exec ./server "$@"
+```
+
+# File structure
+
+- The files for your solution are placed in the `src/` directory.
+- `src/Server.cpp` contains the `main` function, which is what the test runner executes.
+- `CMakeLists.txt` contains the rules for compiling your program.
+
+# Adding more files
+
+You can add more files and directories to `src/`. The test runner will include them when compiling your code.
+
+For example, if you added a file at `src/Foo.hpp`, you could use it like so:
+
+<CodeGroup>
+```cpp src/Foo.hpp
+#pragma once
+
+#include <iostream>
+
+void foo() {
+std::cout << "Hello, World!";
+}
+
+````
+
+```cpp src/Server.cpp
+#include "Foo.hpp"
+
+int main(int argc, char* argv[]) {
+    foo()   // calls foo() defined in src/Foo.hpp
+}
+````
+
+</CodeGroup>
+
+<Note>
+  `#pragma once` is a preprocessor directive that ensures the header file is
+  only included once to prevent any duplication issues.
+</Note>
+
+# Adding dependencies
+
+Adding third-party dependencies to a C++ project involves the following steps:
+
+<Steps>
+  <Step title="Download the library">
+    Download the library files from the source.
+    <Note>
+      Some libraries come built into your operating system and don't need to be
+      installed. E.g. `pthread` comes built-in to Unix systems (Linux and
+      macOS).{" "}
+    </Note>
+  </Step>
+  <Step title="Update your Include Path">
+ Add the following line to your `CMakeLists.txt` file:
+
+```CMake
+include_directories(/path/to/my_library/include)
```

Comment:
@glokta1 let's use a concrete example here, pick something like asio (used in networking challenges a lot). It isn't super clear from this how the user should set the path.

For example, the `/path/...` here suggest that this must be an absolute path, but a user doesn't control the absolute path on our test runners, so how would that work?